x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -1566,9 +1575,15 @@ export class RulesClient {
     return this.spaceId;
   }
 
-  private async scheduleAlert(id: string, alertTypeId: string, schedule: IntervalSchedule) {
-    return await this.taskManager.schedule({
-      taskType: `alerting:${alertTypeId}`,
+  private async scheduleRule(
+    id: string,
+    ruleTypeId: string,
+    schedule: IntervalSchedule,
+    throwOnConflict: boolean // whether to throw conflict errors or swallow them
```

Comment:
When rules are created, `409` conflict errors are thrown and block rule creation since what is happening is that there is already a task document with the same ID as the rule and we don't want to overwrite the existing task document. Since rule IDs are randomly generated, I think the chances of collision are very low.

When rules are enabled, `409` conflict errors are swallowed since this indicates a race condition between enabling and disabling.